Scripts/Diagnostics/Azure_AppService_DP_Use_Secure_FTP_Deployment.ps1

```diff
@@ -0,0 +1,150 @@
+$debug = $true
+
+function Get-AppServiceFtpState()
+{
+  <#
+    .SYNOPSIS
+    This command shows the current FTP state for the App Service Production slot and all non-Production slots.
+    .DESCRIPTION
+    This command shows the current FTP state for the App Service Production slot and all non-Production slots.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the App Service to be remediated.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the App Service.
+    .PARAMETER AppServiceName
+    The App Service name.
+    .INPUTS
+    None
+    .OUTPUTS
+    Dictionary whose keys are App Service slot names, and whose values are the corresponding slot FTP settings.
+    .EXAMPLE
+    PS> Get-AppServiceFtpState -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -AppServiceName "MyAppServiceName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param
+  (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $AppServiceName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $appService = Get-AzWebApp -ResourceGroupName $ResourceGroupName -Name $AppServiceName
+
+  $ftpConfigs = [ordered]@{ "Production" = $appService.SiteConfig.FtpsState }
```

Comment:
Just a thought, should we also add a function to validate basic auth as well?

Reply:
Added a function